### ISSUE TITLE
[Date field] Contrast of text with year range in dark mode is insufficient

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3466]
### 😯 Current Behavior

Select year view in Date field in dark mode. Contrast for this text is 1.4:1. It should be not less than 4.5:1.

<img width="303" height="381" alt="Image" src="https://github.com/user-attachments/assets/f7b1ce83-6044-40b1-bad6-6d034536fb47" />

### 🤔 Expected Behavior/possible solution?

Text should have the same color as it has in Calendar stories: 

<img width="265" height="289" alt="Image" src="https://github.com/user-attachments/assets/22664cef-fbb1-4796-91d7-17210d4b0dfd" />

### 🖥️ Steps to Reproduce

1. Go to https://storybook.sanomalearning.design/?path=/story/form-date-field--basic&globals=mode:dark
2. Select 'dark mode'
3. Open date picker
4. Go to 'Select year' view
5. Check contrast of text with year range

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/form-date-field--basic&globals=mode:dark

### ⚖️ Impact

Not a huge problem, we have a workaround

### Framework

Other

### What browsers are you seeing the problem on?

All

### Component version

0.1.4

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
